test/src/test/java/jenkins/triggers/ReverseBuildTriggerTest.java

```diff
@@ -148,4 +157,30 @@
         assertEquals(new Cause.UpstreamCause((Run) b), downstream.getLastBuild().getCause(Cause.UpstreamCause.class));
     }
 
+    @Issue("JENKINS-29876")
+    @Test
+    public void nullJobInTriggerNotCausesNPE() throws Exception {
+        final FreeStyleProject upstreamJob = r.createFreeStyleProject("upstream");
+
+        //job with trigger.job == null
+        final FreeStyleProject downstreamJob1 = r.createFreeStyleProject("downstream1");
+        final ReverseBuildTrigger reverseBuildTrigger = new ReverseBuildTrigger("upstream", Result.SUCCESS);
+        downstreamJob1.addTrigger(reverseBuildTrigger);
```

Comment:
@jglick filled https://issues.jenkins-ci.org/browse/JENKINS-30802 as it really confuses everybody